Libraries/Components/TextInput/AndroidTextInputNativeComponent.js

```diff
@@ -165,6 +165,19 @@ export type NativeProps = $ReadOnly<{|
     'off',
   >,
 
+  /**
+   * String to be read by screenreaders to indicate an error state. The acceptable parameters
+   * of accessibilityErrorMessage is a string. Setting accessibilityInvalid to true activates
+   * the error message. Setting accessibilityInvalid to false removes the error message.
+   */
+  accessibilityErrorMessage?: ?Stringish,
+
+  /**
+   * Setting accessibilityInvalid to true activates the error message. Setting accessibilityInvalid
+   * to false removes the error message.
+   */
+  accessibilityInvalid?: ?boolean,
+
```

Comment:
What is the rationale for adding two props when they could instead be one? (Also, probably no need to reiterate in the type of the prop in the docblock, since the type is already defined and will be presumably extracted for any reference documentation purposes.)

```suggestion
  /**
   * String to be read by screen readers to indicate an error state. Setting this to null or
   * undefined will remove the error message.
   */
  accessibilityErrorMessage?: ?Stringish,
```

Reply:
>What is the rationale for adding two props when they could instead be one? 

It was introduced with the code review https://github.com/facebook/react-native/pull/33468#issuecomment-1147723451 

Adapts this API to the Web API aria-errormessage
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage

>When there is a user-created error, you want to let them know it exists and tell them how to fix it. There are two attributes you need to use: set [aria-invalid="true"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-invalid) to define the object as being in an error state, then add the aria-errormessage attribute with the value being the id of the element containing the error message text for that object.

This API (`accessibilityErrorMessage` and `accessibilityInvalid`) is based on Android [AccessbilityNodeInfo](https://github.com/aosp-mirror/platform_frameworks_base/blob/00545d0fd6ba561271b690bac51f3439d3bc56e5/core/java/android/view/accessibility/AccessibilityNodeInfo.java#L3005-L3008) API. The API allows adding accessibility information to each View node and announcing it with [talkback](https://github.com/google/talkback).

Each android view and widget (EditText, TextView or View) stores their accessibility information in their AccessibilityNodeInfo node.
TalkBack reads this information when the screenreader's focus changes.

The error message of an EditText is set on the AccessibillityNodeInfo using the following APIs:

- [AccessbilityNodeInfo#setError](https://github.com/aosp-mirror/platform_frameworks_base/blob/00545d0fd6ba561271b690bac51f3439d3bc56e5/core/java/android/view/accessibility/AccessibilityNodeInfo.java#L3005-L3008) to set the error text of the node.
- [AccessibilityNodeInfo#setContentInvalid](https://github.com/aosp-mirror/platform_frameworks_base/blob/00545d0fd6ba561271b690bac51f3439d3bc56e5/core/java/android/view/accessibility/AccessibilityNodeInfo.java#L2502-L2504) to set the content as invalid.  For example, a date is not well-formed.

https://github.com/facebook/react-native/pull/33468#issuecomment-1147732785
>something to note is that the value of aria-errormessage is meant to be the id of an element that contains the error message.

This API relies on the [AccessbilityNodeInfo#setError](https://github.com/aosp-mirror/platform_frameworks_base/blob/00545d0fd6ba561271b690bac51f3439d3bc56e5/core/java/android/view/accessibility/AccessibilityNodeInfo.java#L3005-L3008). It does not add a hidden text element to the View hierarchy using react-native. Instead, we use the Native Android API (AccessibilityNodeInfo) to add the error message to the node of the EditText. TalkBack announces the error message when focusing on the EditText (TextInput).

I'll be happy to remove the accessibililtyInvalid prop and further improve the PR. Thanks

Reply:
> no need to reiterate in the type of the prop in the docblock, since the type is already defined and will be presumably extracted for any reference documentation purposes.

I removed the documentation in the docblock of prop accessibilityInvalid with commit https://github.com/facebook/react-native/pull/33468/commits/cb0c5060cb721974f7d39bf3fb2fa2cecb2bc7f1

Reply:
Hmm this is interesting because it doesn't completely follow the spec since `accessibilityErrorMessage` doesn't provide the elementId but it also introduces this extra logic around `accessibilityInvalid` which is part of the web standards. 

I think I'd lean towards keeping the two separate props but including in the documentation that we're modelling this after `aria` standards. See my comment here: https://github.com/facebook/react-native-website/pull/3010#discussion_r957610518

Reply:
Thanks a lot, @lunaleaps. I updated the documentation as requested with commit  https://github.com/facebook/react-native-website/pull/3010/commits/8510e8cba503d523f18053e7e2c3b3125f11c3b6  🙏 .

<details><summary>preview of the documentation</summary>
<p>

![Screen Shot 2022-08-30 at 21 03 45](https://user-images.githubusercontent.com/24992535/187443887-a4cda43d-cfc0-452e-8ad4-ecec6008b678.png)

</p>
</details>